sentry-core/src/test/java/io/sentry/transport/HttpTransportTest.kt

```diff
@@ -0,0 +1,130 @@
+package io.sentry.transport
+
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.eq
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import io.sentry.ISerializer
+import io.sentry.SentryEvent
+import io.sentry.SentryOptions
+import java.io.IOException
+import java.net.HttpURLConnection
+import java.net.Proxy
+import java.net.URI
+import java.net.URL
+import java.net.URLConnection
+import java.util.function.Consumer
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+
+class HttpTransportTest {
+
+    private class Fixture {
+        var dsn = URI.create("http://localhost").toURL()
+        val serializer = mock<ISerializer>()
+        var proxy: Proxy? = null
+        var requestUpdater: Consumer<URLConnection> = Consumer { ; }
+        var connectionTimeout = 1000
+        var readTimeout = 500
+        var bypassSecurity = false
+        val connection = mock<HttpURLConnection>()
+
+        init {
+            whenever(connection.outputStream).thenReturn(mock())
```

Comment:
would it not be better `@BeforeTest`?

Reply:
I thought the whole `Fixture` thing is a kind of before-test. But I will adapt to whatever style you prefer.

Reply:
Fixture tests are a fixed state, but if we need to extend those tests and somehow need the state of the mocks to be cleaned, the testing lifecycle @Before/After would be better

Reply:
JUnit4 does create a new instance for each test run I think.

Reply:
it does, if we don't wanna play with the values of the `Fixture`, all good.
if somehow we would like to change some values of the `Fixture` and run tests in parallel/out of order, might find flaky tests, so in that case, `@Before` will make the setup and clean the states each test, but again, if it's not the case in this class, I'm happy to keep it like this

Reply:
So let's leave this in and fix it if we ever find a problem you describe (which I think would only ever manifest if we changed the test instantiation to be per-class, not per test-method, and we had parallel test runs, which I know is possible with JUnit5 but I am not sure about JUnit4). But IMHO in the case of per-class instantiation and parallel test invocations, a @Before method doesn't make a difference - you'd still be mutating shared data.

I will try to read up on the fixture approach because frankly it's new to me.